test/apps/iodemo/io_demo.cc

```diff
@@ -448,6 +450,11 @@ class P2pDemoCommon : public UcxContext {
             case UCS_MEMORY_TYPE_HOST:
                 free(_buffer);
                 break;
+            case UCS_MEMORY_TYPE_UNKNOWN:
+                /* If the buffer was used outside am recv callback,
+                 * it should be freed here by calling ucp_am_data_release().
+                 */
+                break;
```

Comment:
it's a hack to use memory type as indication of whether need to free the buffer or not

Reply:
UCS_MEMORY_TYPE_UNKNOWN indicates external buffers (i.e. buffers allocated by UCX) it doesn't indicate when to free external buffers. the comment here was to remind users of freeing external buffers if they have used them outside am recv callback. Maybe a better place to free external buffers is Buffer::release().

Reply:
UCS_MEMORY_TYPE_UNKNOWN is not an indication of how the buffer is allocated, so it's not right to use it for this purpose

Reply:
keep Buffer class as it was. Please check the latest commit.